Dev to Main

kevinicolas22 opened the pull request:
Foram adicionadas a branch dev: landing page, login page (para aluno e para professor) e cadastro page (só pode ser acessada pela página do login de aluno, visto que alguns campos só fazem sentido para aluno, como o "curso" e  "matrícula", e para que isso funcione, adicionei um evento onclick, ao passar o mouse pela frase "Ainda não tenho cadastro"). Todas as páginas estão responsivas.

LeilaFarias committed:
feat: add componente input email e variações

magnofilipe committed:
feat: Prime React library added to project

magnofilipe committed:
feat: 3 new buttons

4nandaw committed:
feat: password input

LeilaFarias committed:
fix: substitui px por vh e vw

LeilaFarias committed:
fix: color input

LeilaFarias committed:
add: directory inputEmail

4nandaw committed:
fix: showToggle true as default

magnofilipe committed:
Merge branch 'input-email' into feat/login-screen

magnofilipe committed:
Merge remote-tracking branch 'origin/input-email' into feat/login-screen

magnofilipe committed:
Merge remote-tracking branch 'origin/password-input' into feat/login-screen

magnofilipe committed:
fix: PasswordInput directory position files + package-lock.json conflicts

magnofilipe committed:
feat: new font

Co-authored-by: Kevin Nicolas <kevinicolas22@users.noreply.github.com>

magnofilipe committed:
feat: login screen. more adjustments are needed

magnofilipe committed:
fix: refactor components to login sceen. more adjustments are needed.

kevinicolas22 committed:
add: landing page

4nandaw committed:
feat: cadastro page

Co-authored-by: Leila Farias <LeilaFarias@users.noreply.github.com>

LeilaFarias committed:
feat: add error message and InputTemplate

LeilaFarias committed:
fix: adjust layout and add responsiveness

LeilaFarias committed:
chore: adjust button css

kevinicolas22 committed:
refactor: add responsive to login page and apply login page for student and external users

kevinicolas22 committed:
Merge branch 'feat/landing-page' into dev

kevinicolas22 committed:
Merge branch 'feat/cadastro-screen' into dev

kevinicolas22 committed:
refactor: add link to cadastro page

LeilaFarias commented on src/pages/LoginAluno.jsx:
```diff
@@ -11,14 +14,35 @@ function LoginAluno() {
+                        <h1>Faça login no sistema</h1>
+                        <form onSubmit={handleSubmit}>
+                            <EmailInput />
+                            <PasswordInput />
```
esses componentes estao diferentes um do outro

LeilaFarias commented on src/pages/Home.jsx:
```diff
@@ -1,18 +1,287 @@
+                        organizar suas atividades esportivas.
+                    </p>
+                    <div className="cta-button-container">
+                        <a href="index.html" className="btn-cta">
```
```suggestion
                        <a href="/cadastro" className="btn-cta">
```

LeilaFarias commented on src/pages/LoginAluno.jsx:
```diff
@@ -11,14 +14,35 @@ function LoginAluno() {
+                <div className="left-section">
+                    <div className="left-section-elements">
+                        <img src="src/assets/Logo_3.png" alt="Logo" />
+                        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua</p>
```
precisamos mudar este texto

LeilaFarias commented on src/pages/LoginProfessor.jsx:
```diff
@@ -11,14 +14,32 @@ function LoginProfessor() {
+                            <DefaultButton label={"Entrar"}></DefaultButton>
+                        </form>
+                        <p>Esqueci minha senha </p>
+                        <p>Ainda não tenho cadastro</p>
```
```suggestion
<p onClick={() => navigate("/cadastro")} style={{ cursor: "pointer" }}>
        Ainda não tenho cadastro
</p>   
```

LeilaFarias commented on src/pages/components/Password/PasswordInput.jsx:
```diff
@@ -0,0 +1,28 @@
+  return (
+    <div className="password-container">
+      <input
+        type={showPassword ? "text" : "password"}
```
```suggestion
       type={showPassword ? "password" : "text"}
```
o "olhinho" esta ao contrario, quando ele esta aberto a senha fica invisivel e quando esta fechado a senha fica visivel, mas devia ser o contrario

LeilaFarias commented on src/pages/components/Password/PasswordInput.jsx:
em determinado tamanho de telas o toggle "entra" pra dentro do input 

![image](https://github.com/user-attachments/assets/b0bb430b-28bf-4ccc-8d4a-cea573d1822c)

kevinicolas22 replied:
Resolved

LeilaFarias commented on src/pages/Cadastro.css:
```diff
@@ -0,0 +1,115 @@
+  font-size: 40px;
+  margin-bottom: 25px;
+}
+
```
```suggestion
.arrow-left {
  cursor: pointer;
}
```

LeilaFarias commented on src/pages/components/InputEmail/InputEmail.jsx:
acho q esse InputEmail não é mais necessário, ele pode ser substituido pelo InputTemplate tranquilamente

LeilaFarias commented on src/pages/Home.jsx:
```diff
@@ -1,18 +1,287 @@
+                        </div>
+                    </div>
+                    <div className="nav-right">
+                        <Link to="/login-aluno" className="btn-aluno">
+                            Entrar como Aluno
+                        </Link>
+                        <Link to="/login-professor" className="btn-externo">
+                            Entrar como Externo
+                        </Link>
```
no mobile esse layout da uma quebrada 
![image](https://github.com/user-attachments/assets/d1195194-62ae-4c88-8ad6-66b8e376f942)

kevinicolas22 replied:
resolved

LeilaFarias commented on src/pages/LoginAluno.jsx:
```diff
@@ -11,14 +14,35 @@ function LoginAluno() {
+                            <PasswordInput />
+                            <DefaultButton label="Entrar" />
+                        </form>
+                        <p>Esqueci minha senha</p>
+                        <p onClick={() => navigate("/cadastro")} style={{ cursor: "pointer" }}>
+                            Ainda não tenho cadastro
```
no mobile, esses paragrafos ficam um tanto pequenos comparado ao resto
![image](https://github.com/user-attachments/assets/9d8856e2-0193-4977-8a32-6eb2172e10e4)

LeilaFarias reviewed (commented):
Boa implementação no geral, comentei alguns pontos que podemos melhorar e alguns detalhes que passaram despercebidos

kevinicolas22 committed:
Update src/pages/Home.jsx

fix: leila suggestions

Co-authored-by: Leila Farias <115854945+LeilaFarias@users.noreply.github.com>

kevinicolas22 committed:
Update src/pages/LoginProfessor.jsx

fix: leila suggestions

Co-authored-by: Leila Farias <115854945+LeilaFarias@users.noreply.github.com>

kevinicolas22 committed:
Update src/pages/components/Password/PasswordInput.jsx

fix: leila suggestions

Co-authored-by: Leila Farias <115854945+LeilaFarias@users.noreply.github.com>

kevinicolas22 committed:
Update src/pages/Cadastro.css

fix: leila suggestions

Co-authored-by: Leila Farias <115854945+LeilaFarias@users.noreply.github.com>

kevinicolas22 committed:
fix: resolved leila suggestions

LeilaFarias committed:
fix: email inputs

LeilaFarias reviewed (commented):
Notei mais alguns pontos que podemos revisar, são apenas sugestões, podemos discutir caso notem um conflito com os requisitos

- Centralizar este "entrar como Aluno", no mobile funciona bem porém no desktop comum ele não está centralizado.
Inclusive, podemos mudar para "Entrar como usuário UFCG" ou "Entrar como UFCG", algo do tipo
![image](https://github.com/user-attachments/assets/5d97da0d-bd75-4ca9-bbfb-b4cfbb13344e)

- Mudar a URL para /login-ufcg, inclusive refletir essas mudanças no resto do código, onde tem "login-aluno" substituir para "login-ufcg"
![image](https://github.com/user-attachments/assets/5e78df00-2c7d-40c8-9666-cc57a27eb0ba)

- Aqui da mesma forma, mudar para /login-externo e refletir essa mudança no resto do código
![image](https://github.com/user-attachments/assets/a049063e-25a2-4664-bb57-04c5f39e0533)

kevinicolas22 committed:
add: cadastrar reserva page

kevinicolas22 committed:
refactor: adjusting components

kevinicolas22 committed:
fix: adjusting css details

kevinicolas22 committed:
fix: adjusting comments

magnofilipe committed:
fix: align text to start in "feature-card" class

magnofilipe committed:
fix: gap between two components in css
feat: add border image and modify shadows to better styling

magnofilipe committed:
feat: new component: SecondaryButton

magnofilipe committed:
fix: wrong css styles and missing using SecondaryButton component

magnofilipe committed:
fix: remove logo from Home

magnofilipe committed:
feat: add useNavigate() to all Buttons to make them more reusable

magnofilipe committed:
fix: missing routes to buttons in Home.jsx

magnofilipe committed:
fix: improve footer style by updating data and adding padding

magnofilipe committed:
fix: "Cpf" renamed to "CPF" in InputTemplate component placeholder

magnofilipe committed:
fix: style diffs between Inputs + PasswordInput incorrect width

magnofilipe committed:
fix: missing texts in left-section describing our project Racha10

magnofilipe committed:
fix: incorrect colors from gradient on Home.css

magnofilipe reviewed (changes requested):
Realizei uma série de modificações para que o merge do Pull Request seja bem-sucedido:

1. Correções de estilo, como alinhamento de textos, _paddings_, escritas errôneas (typos) e cores erradas;
2. Adicionado novo componente: SecondaryButton. O mesmo é utilizado na "Home" e substitui o botão improvisado que estava lá anteriormente;
3. Agora, todos os botões possuem _useNavigate()_ em suas codificações, o que permite maior reusabilidade, porque agora aceitam funções (_onClick_) ou redirecionamentos de rotas (_to_);

**Todas as modificações de estilo foram realizadas, contudo, algumas páginas ainda não estão responsivas. Precisamos que todas fiquem!**

kevinicolas22 committed:
fix: media queries (landing page)

kevinicolas22 committed:
fix: adjusting names of the div classes

kevinicolas22 committed:
Merge branch 'origin/feat/cadastrar-reserva' into dev

magnofilipe committed:
fix: responsive failures in Home.jsx
feat: Menubar component to use in Home.jsx.
fix: removed wrong logo design in Home.jsx.

magnofilipe committed:
feat: add Menubar box-shadow in .css to better style

magnofilipe committed:
fix: button paddings

magnofilipe committed:
fix: responsive fixes in Login pages (LoginAluno and LoginProfessor)

kevinicolas22 committed:
fix: adjusting CSS details (Cadastrar Reserva)

kevinicolas22 committed:
feat: header component

kevinicolas22 committed:
add: vizualizar-reserva and add components: MainContent and Header
